src/checkers/inference/model/ConstraintManager.java

```diff
@@ -37,171 +30,186 @@
 
     private VisitorState visitorState;
 
-    private ConstraintVerifier constraintVerifier;
-
     public void init(InferenceAnnotatedTypeFactory inferenceTypeFactory) {
         this.inferenceTypeFactory = inferenceTypeFactory;
         this.realQualHierarchy = inferenceTypeFactory.getRealQualifierHierarchy();
         this.visitorState = inferenceTypeFactory.getVisitorState();
         this.checker = inferenceTypeFactory.getContext().getChecker();
-        this.constraintVerifier = new ConstraintVerifier(realQualHierarchy);
     }
 
     public Set<Constraint> getConstraints() {
         return constraints;
     }
 
-    public ConstraintVerifier getConstraintVerifier() {
-        return constraintVerifier;
-    }
-
-    private void add(Constraint constraint) {
-        if (!ignoreConstraints) {
+    /**
+     * Checks to see if the given {@link Constraint} is an instance of
+     * {@link AlwaysFalseConstraint}. If so, a warning is issued. If not, it adds the
+     * {@link Constraint} to the constraint set only if the {@code ignoreConstraints} flag is set to
+     * false and the constraint is not an {@link AlwaysTrueConstraint}.
+     *
+     * <p>
+     * The source constraint class is used to construct an error key for issue the warning.
+     *
+     * @param constraint a (possibly normalized) constraint
+     * @param nonNormalizedConstraintClass the class literal of the original non-normalized
+     *        constraint
+     */
+    public void add(Constraint constraint,
+            Class<? extends Constraint> nonNormalizedConstraintClass) {
+        if (constraint instanceof AlwaysFalseConstraint) {
+            // converts "XxxConstraint" into "xxx.constraint.unsatisfiable" for use as an error key
+            String errorKey = nonNormalizedConstraintClass.getSimpleName().toLowerCase()
+                    .replace("constraint", ".constraint.unsatisfiable");
+
+            // issue a non-halting error message indicating which non-normalized constraint cannot
```

Comment:
Split error reporting back into individual addXXX methods

Reply:
Add messages.properties file to expand the error message keys into full english. Place in inferenceChecker's directory?   It is any directory up the supertype chain from a specific inference checker.